### PR TITLE
JS-162 Create API module

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: sonar-plugin/sonar-javascript-plugin/pom.xml [hunk 1]
```diff
@@ -26,16 +26,20 @@
...
     <dependency>
       <groupId>${project.groupId}</groupId>
-      <artifactId>javascript-checks</artifactId>
+      <artifactId>bridge</artifactId>
     </dependency>
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>css</artifactId>
     </dependency>
+    <dependency>
```
Comment (saberduck) [inline sonar-plugin/sonar-javascript-plugin/pom.xml:39]: reordered to follow alphabetical order

File: sonar-plugin/api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  </parent>
+
+  <artifactId>api</artifactId>
+
+  <name>SonarQube JavaScript :: API</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.sonarsource.api.plugin</groupId>
+      <artifactId>sonar-plugin-api</artifactId>
```
Comment (ilia-kebets-sonarsource) [inline sonar-plugin/api/pom.xml:18]: shouldn't it depend on bridge instead?

**Review by ilia-kebets-sonarsource — CHANGES_REQUESTED** [review]
see comment

File: sonar-plugin/api/pom.xml (hunk 2 above)
Comment (saberduck) [reply sonar-plugin/api/pom.xml:18]: why? This means it depends on Sonar (scanner) API; it's to have access to `InputFile,` `@ScannerSide,` etc...

**Review by ilia-kebets-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**